### PR TITLE
Do not encode the server port in the test file

### PR DESCRIPTION
**Pull request by keertip**
Remove the encoding the port information for the web socket connection at test file generation time,  instead set it as an environment variable. This allows for the test file to be created and kernel generated without any test harness specific information.

### CODE REVIEW

**Review by tvolkert — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -550,19 +552,20 @@ class _FlutterPlatform extends PlatformPlugin {
...
 
   String _generateTestMain({
     String testUrl,
     String encodedWebsocketUrl,
+    String serverPort,
   }) {
     return '''
 import 'dart:convert';
-import 'dart:io'; // ignore: dart_io_import
+import 'dart:io';
```
[the remark this answers is not in the document]
Comment (keertip) [reply packages/flutter_tools/lib/src/test/flutter_platform.dart]: done.

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 2]
```diff
@@ -550,19 +552,20 @@ class _FlutterPlatform extends PlatformPlugin {
...
-        ? 'ws://${host.address}:${server.port}'
-        : 'ws://[${host.address}]:${server.port}';
+        ? 'ws://${host.address}'
+        : 'ws://[${host.address}]';
   }
 
   String _generateTestMain({
     String testUrl,
     String encodedWebsocketUrl,
+    String serverPort,
```
[the remark this answers is not in the document]
Comment (keertip) [reply packages/flutter_tools/lib/src/test/flutter_platform.dart]: If I remove this, _generateTestMain tries to find a value to substitute for '$serverPort' and fails -Class '_FlutterPlatform' has no instance getter 'serverPort'.

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 3]
```diff
@@ -576,7 +578,8 @@ import '$testUrl' as test;
 
 void main() {
   print('$_kStartTimeoutTimerMessage');
-  String server = Uri.decodeComponent('$encodedWebsocketUrl');
+  String serverPort = Platform.environment['SERVER_PORT'];
+  String server = Uri.decodeComponent('$encodedWebsocketUrl:$serverPort');
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Just escape the dollar - `\$serverPort`.  There's no need to have it as a param to `_generateTestMain()`

Comment (keertip) [reply packages/flutter_tools/lib/src/test/flutter_platform.dart]: done

**Review by tvolkert — APPROVED** [review]

**Comment by Hixie** [comment]
BTW, this was checked in when the tree was red. Please avoid checking anything in when the tree is red except for attempts to fix the tree, because it makes it much harder to detect, and later identify, additional failures introduced while the previous failure was ongoing. If you suspect a flaky result, please re-run the test rather than checking in while red. If that isn't possible, at a minimum please note the situation in the PR when landing it. Thanks!

**Comment by keertip** [comment]
Did look at the failure, it was on a travis bot and the dart2 tests, and for the PR is looked all green. Found out now that there are other tests that are run on master, so my reading may not be right. Sorry about that.

**Pull request merged**